### PR TITLE
feat: enable make help for the top level makefile

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- This is an adhoc implementation to support the `make help` target for the top level Makefile.
- There is already a python script to allow to do something similar but it seems it does not support the current docs in the toplevel Makefile. https://github.com/elastic/beats/blob/8d732882c89ad557e3b056c1a839ac87b2602881/libbeat/scripts/Makefile#L415-L419

## Tasks
- [ ] Agree whether this is a valid use case
- [ ] Document the `TBD` targets.

## Output

```
make help
Usage: make [target] [VARIABLE=value]
Targets:
  help                       Show this help.
  testsuite                  Runs complete testsuites (unit, system, integration) for all beats with coverage and race detection. Also it builds the docs and the generators.
  setup-commit-hook          Setup the git pre-commit hook
  stop-environments          Stop the environment for each project.
  test                       Runs unit and system tests without coverage and race detection.
  unit                       Runs unit tests without coverage and race detection.
  crosscompile               Crosscompile all beats.
  coverage-report            Generates coverage report.
  update                     TBD.
  clean                      Clean target.
  clean-vendor               Cleans up the vendor directory from unnecessary files. This should always be run after updating the dependencies
  check                      TBD.
  check-headers              Check the license headers.
  add-headers                Adds the license headers.
  misspell                   Corrects spelling errors.
  fmt                        TBD.
  lint                       TBD.
  docs                       Builds the documents for each beat
  notice                     Generates the NOTICE file.
  python-env                 Sets up the virtual python environment.
  test-apm                   Tests if apm works with the current code
  snapshot                   Builds a snapshot release.
  release                    Builds a release.
  release-manager-snapshot   Builds a snapshot release. The Go version defined in .go-version will be installed and used for the build.
  release-manager-release    Builds a snapshot release. The Go version defined in .go-version will be installed and used for the build.
  beats-dashboardse          Collects dashboards from all Beats and generates a zip file distribution.
Variables:
  BEATS                      Default values: auditbeat filebeat heartbeat journalbeat metricbeat packetbeat winlogbeat x-pack/functionbeat
  PYTHON_ENV                 Default values: $(BUILD_DIR)/python-env
  VIRTUALENV_PARAMS          Default values: 
  REVIEWDOG_OPTIONS          Default values: -diff "git diff master"

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -113,69 +127,70 @@ misspell:
...
 		-exec misspell -w {} \;
 
+## fmt : TBD.
 .PHONY: fmt
 fmt: add-headers python-env
 	@$(foreach var,$(PROJECTS) dev-tools $(PROJECTS_XPACK_MAGE),$(MAKE) -C $(var) fmt || exit 1;)
 	@# Cleans also python files which are not part of the beats
 	@$(FIND) -name "*.py" -exec $(PYTHON_ENV)/bin/autopep8 --in-place --max-line-length 120 {} \;
 
+## lint : TBD.
```
Comment (cachedout) [inline Makefile]: Does this need to be `TBD`?

File: Makefile [hunk 2]
```diff
@@ -113,69 +127,70 @@ misspell:
...
 
-# Builds a snapshot release. The Go version defined in .go-version will be
-# installed and used for the build.
+## release-manager-release : Builds a snapshot release. The Go version defined in .go-version will be installed and used for the build.
 .PHONY: release-manager-release
 release-manager-release:
 	./dev-tools/run_with_go_ver $(MAKE) release
 
-# Collects dashboards from all Beats and generates a zip file distribution.
+## beats-dashboardse : Collects dashboards from all Beats and generates a zip file distribution.
```
Comment (cachedout) [inline Makefile]:
```suggestion
## beats-dashboards : Collects dashboards from all Beats and generates a zip file distribution.
```

File: Makefile (hunk 1 above)
Comment (v1v) [reply Makefile]: There was no description, so I put TBD to agree whether we wanted to add something or not. I created one task to confirm with everyone what's the strategy. No strong opinion about what's to do.

Comment (cachedout) [reply Makefile]: Ah, gotcha. Makes sense. 👍

**Pull request merged**